core/cc/dl_loader.h

```diff
@@ -22,10 +22,12 @@ namespace core {
 // Utility class for retrieving function pointers from dynamic libraries.
 class DlLoader {
 public:
-    // Loads the specified dynamic library.
-    // If the library cannot be loaded then this is a fatal error.
-    // For *nix systems, a nullptr can be used to search the application's functions.
-    DlLoader(const char* name);
+    // Loads the dynamic library specified by the given name and fallback names
+    // (if any). Names will be used to try to find the library in order. If the
+    // library cannot be loaded then this is a fatal error. For *nix systems,
+    // a nullptr can be used to search the application's functions.
+    template<typename... ConstCharPtrs>
```

Comment:
Is there a particular reason we are doing this with template magic, rather than something like std::initializer_list?

Reply:
hmm, I just feel `DlLoader("libvulkan.so")` looks a little bit better than `DlLoader({"libvulkan.so"})`. But yes, initializer list can do the same and does not need explicit instantiation...Don't have other particular reasons.

Reply:
Ok, as much as I personally love templates, reading

```
void* load() {
    return nullptr;
}

template<typename... ConstCharPtrs>
void* load(const char* name, ConstCharPtrs... fallback_names) {
....
...
    return load(fallback_names...);
...
...
Bunch of macros that will cause a link error if we hit 6 arguments.
...
```
is a lot harder to parse than
```
void* load(std::initializer_list<const char* name> names) {
   for (auto& nm: names) {
   }
```